### DISCUSSION TITLE
Can't add multiple M2O fields with the same parent and child

### DISCUSSION THREAD

**Discussion by kamalseisy** [directus/directus#16620] [Archive]
### Describe the Bug

if I have two collections (parent_collection and child_collection) and want to add 2 fields to the child collection related to the parent collection with M2O relation, the second field will not be added
for example if I want to add mother_id and father_id to the child_collection both fields are foreign keys to parent_collection, mother_id as M2O field will be added successfuly but adding father_id will add string field (not M2O) and show error message

![image](https://user-images.githubusercontent.com/10973700/201516401-70174b9a-abf6-4e79-83d4-c5ea88dfa515.png)

### To Reproduce

1. create collection "parent_collection"
2. create collection "child_collection"
3. add M2O field mother_id to child_collection related to parent_collection
4. add M2O field father_id to child_collection related to parent_collection

### Errors Shown

ALTER TABLE [child_collection] ADD CONSTRAINT [child_collection_father_id_foreign] FOREIGN KEY ([father_id]) REFERENCES [parent_collection] ([id]) ON DELETE SET NULL - 

### What version of Directus are you using?

9.20.4

### What version of Node.js are you using?

18.4.0

### What database are you using?

Microsoft SQL Server

### What browser are you using?

Chrome

### How are you deploying Directus?

running locally

**Comment by rijkvanzanten** [comment]
I believe this is because MS SQL doesn't support having more than one foreign key constraint with a trigger to the same table, so in this case `mother_id` and `father_id` are both a foreign key with a constraint to the same parent, which isn't allowed in MSSQL 🤔

**Comment by licitdev** [comment]
Can confirm that this is MSSQL specific which occur even when manually creating the fields in the database, with constraint actions. 

To resolve, the relational trigger for `mother_id` field should be updated to `Prevent the deletion`, in order to add the `father_id` M2O field. This will in turn set the foreign key constraint to `NO ACTION`.

Alternatively, set the relational trigger to be `Prevent the deletion` in Advance Field Creation Mode when creating `father_id`.

https://user-images.githubusercontent.com/26413686/203996807-18193891-42cb-4209-aae8-d415a1ac4f2f.mov
